translatable_shared/src/translations/node.rs

```diff
@@ -172,36 +172,85 @@ impl ToTokens for TranslationNode {
 impl TryFrom<Table> for TranslationNode {
     type Error = TranslationNodeError;
 
-    // The top level can only contain objects is never enforced.
+    /// Attempts to convert a `Table` into a `TranslationNode`.
+    ///
+    /// This function iterates over the key-value pairs in the given `Table`.
+    /// Depending on the type of each value, it either adds a translation to
+    /// a `TranslationObject` or a nested `TranslationNode` to a `TranslationNesting`.
+    /// If the conversion fails due to mixed values or invalid nesting, it returns
+    /// a `TranslationNodeError`.
+    ///
+    /// # Arguments
+    ///
+    /// * `value` - A TOML table to be converted.
+    ///
+    /// # Returns
+    ///
+    /// A `Result` containing a `TranslationNode` if successful, or a
+    /// `TranslationNodeError` if the conversion fails.
     fn try_from(value: Table) -> Result<Self, Self::Error> {
         let mut result = None;
 
-        for (key, value) in value {
+        for (key, value) in value.iter() {
```

Comment:
I don't think adding `.iter()` is actually useful, `.iter()` is implictly called on any iterator, you can just make it `ref` or `&`.